osu.Game/Rulesets/UI/DrawableRuleset.cs

```diff
@@ -268,6 +268,12 @@ public bool RemoveHitObject(TObject hitObject)
             return false;
         }
 
+        public override void EndRecording()
+        {
+            if ((KeyBindingInputManager is IHasRecordingHandler recordingInputManager))
```

Comment:
one set of parens too many

```suggestion
            if (KeyBindingInputManager is IHasRecordingHandler recordingInputManager)
```